### DISCUSSION TITLE
can't start up after version 4.13

### DISCUSSION THREAD

**Discussion by daizhuo** [smartavionics/Cura#160] [Q&A]
I can start up with [Cura-mb-master-armhf-20211119.AppImage].
but can't open with[Cura-mb-master-armhf-20220111.AppImage] and higher version.
why?

I have update my raspberry pi 4b to the latest Pi OS version and install all requried apts.

execute in terminal will display:
Found libGLESv2
/tmp/ .mount_cura-mVRgx80/usr/bin//cura.sh: line 36: message: command not found
 <frozen zipimport>259: DeprecationWarning: the imp module is deprecated in favour of import lib; see the module's documentation for alternative uses

I'm a new bird to raspberry pi 4b and just use it for my 3D printing, I don't know any programming too.
Thank you!

**Comment by smartavionics** [comment]
Hi, do you get anything in the stdout.log and stderr.log files that should be created in ~/.local/share/cura or the cura.log file from ~/.local/share/cura/master ?

**Reply by daizhuo** [reply]
thank you so much for your reply.
i got these in stderr.log but nothing in stdout.log.


Fatal Python error: Aborted

Current thread 0xb6f6ac40 (most recent call first):
  File "/home/markb/cura/cura-build/build/inst/lib/python3/dist-packages/UM/Qt/QtApplication.py", line 99 in __init__
  File "/home/markb/cura/cura-build/build/inst/lib/python3/dist-packages/cura/CuraApplication.py", line 153 in __init__
  File "/home/markb/cura/cura-build/build/inst/bin/cura", line 271 in <module>
  File "/opt/cura/lib/python3.8/site-packages/cx_Freeze/initscripts/Console.py", line 36 in run
  File "/opt/cura/lib/python3.8/site-packages/cx_Freeze/initscripts/__startup__.py", line 74 in run

**Comment by smartavionics** [comment]
Hmm, that doesn't tell me much. Please now try this. Run the appimage from a terminal window specifying the --appimage-mount option like this:

./Cura-mb-master-armhf-20220129.AppImage --appimage-mount

It should then print a line that looks like this...

/tmp/.mount_Cura-mUtQsHd

The actual name will be different, just use what it prints in the following example.

You then need to run this command (substituting the name printed by --appimage-mount)...

ldd /tmp/.mount_Cura-mUtQsHd/usr/bin/qt/plugins/platforms/libqxcb.so

It should then print a load of lines, please copy that output to here. Thanks.

**Reply by daizhuo** [reply]
pi@raspberrypi:~ $ cd Downloads
pi@raspberrypi:~/Downloads $ ./Cura-mb-master-armhf-20220129.AppImage --appimage-mount
/tmp/.mount_Cura-mP2KM87
ldd /tmp/.mount_Cura-mP2KM87/usr/bin/qt/plugins/platforms/libqxcb.so

nothing happened after these commands T_T

**Comment by daizhuo** [comment]
pi@raspberrypi:~ $ cd Downloads
pi@raspberrypi:~/Downloads $ ./Cura-mb-master-armhf-20220129.AppImage --appimage-mount
/tmp/.mount_Cura-mP2KM87
ldd /tmp/.mount_Cura-mP2KM87/usr/bin/qt/plugins/platforms/libqxcb.so


nothing happened after these commands T_T

**Reply by smartavionics** [reply]
You need to enter the ldd command in another terminal window because the first one is still doing the mount.

**Reply by daizhuo** [reply]
the first terminal:
pi@raspberrypi:~/Downloads $ ./Cura-mb-master-armhf-20220129.AppImage --appimage-mount
/tmp/.mount_Cura-muy9AHB

the second terminal:
pi@raspberrypi:~ $ ldd /tmp/.mount_Cura-muy94AHB/usr/bin/qt/plugins/platforms/libqxcb.so
ldd: /tmp/.mount_Cura-muy94AHB/usr/bin/qt/plugins/platforms/libqxcb.so: No such file or directory

**Reply by smartavionics** [reply]
So what is in that directory? Do "ls /tmp/.mount_Cura-muy94AHB/usr/bin/qt/plugins/platforms" . Maybe you need to stop the mount and redo it? Remember, every time you mount the appimage the name will change.

**Reply by daizhuo** [reply]
Thank you so much for ur help!
Finally I got these information.

pi@raspberrypi:~ $ ldd /tmp/.mount_Cura-mEqpI3J/usr/bin/qt/plugins/platforms/libqxcb.so
	linux-vdso.so.1 (0xbeedf000)
	/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so => /usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so (0xb6f8d000)
	libQt5XcbQpa.so.5 => /tmp/.mount_Cura-mEqpI3J/usr/bin/qt/plugins/platforms/libQt5XcbQpa.so.5 (0xb6e7a000)
	libfontconfig.so.1 => /tmp/.mount_Cura-mEqpI3J/usr/bin/qt/plugins/platforms/libfontconfig.so.1 (0xb6e30000)
	libfreetype.so.6 => /lib/arm-linux-gnueabihf/libfreetype.so.6 (0xb6d77000)
	libz.so.1 => /lib/arm-linux-gnueabihf/libz.so.1 (0xb6d4c000)
	libQt5Gui.so.5 => /lib/arm-linux-gnueabihf/libQt5Gui.so.5 (0xb68cc000)
	libQt5DBus.so.5 => /lib/arm-linux-gnueabihf/libQt5DBus.so.5 (0xb6833000)
	libQt5Core.so.5 => /lib/arm-linux-gnueabihf/libQt5Core.so.5 (0xb63b9000)
	libGLESv2.so.2 => /lib/arm-linux-gnueabihf/libGLESv2.so.2 (0xb639a000)
	libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb6370000)
	libX11-xcb.so.1 => /lib/arm-linux-gnueabihf/libX11-xcb.so.1 (0xb635e000)
	libxcb-icccm.so.4 => /lib/arm-linux-gnueabihf/libxcb-icccm.so.4 (0xb634a000)
	libxcb-image.so.0 => /lib/arm-linux-gnueabihf/libxcb-image.so.0 (0xb6336000)
	libxcb-shm.so.0 => /lib/arm-linux-gnueabihf/libxcb-shm.so.0 (0xb6323000)
	libxcb-util.so.0 => /lib/arm-linux-gnueabihf/libxcb-util.so.0 (0xb630e000)
	libxcb-keysyms.so.1 => /lib/arm-linux-gnueabihf/libxcb-keysyms.so.1 (0xb62fb000)
	libxcb-randr.so.0 => /lib/arm-linux-gnueabihf/libxcb-randr.so.0 (0xb62de000)
	libxcb-render-util.so.0 => /lib/arm-linux-gnueabihf/libxcb-render-util.so.0 (0xb62ca000)
	libxcb-render.so.0 => /lib/arm-linux-gnueabihf/libxcb-render.so.0 (0xb62af000)
	libxcb-shape.so.0 => /lib/arm-linux-gnueabihf/libxcb-shape.so.0 (0xb629c000)
	libxcb-sync.so.1 => /lib/arm-linux-gnueabihf/libxcb-sync.so.1 (0xb6286000)
	libxcb-xfixes.so.0 => /lib/arm-linux-gnueabihf/libxcb-xfixes.so.0 (0xb626f000)
	libxcb-xinerama.so.0 => /lib/arm-linux-gnueabihf/libxcb-xinerama.so.0 (0xb625c000)
	libxcb-xkb.so.1 => /lib/arm-linux-gnueabihf/libxcb-xkb.so.1 (0xb6237000)
	libxcb-xinput.so.0 => not found
	libxcb.so.1 => /lib/arm-linux-gnueabihf/libxcb.so.1 (0xb6208000)
	libXext.so.6 => /lib/arm-linux-gnueabihf/libXext.so.6 (0xb61e9000)
	libX11.so.6 => /lib/arm-linux-gnueabihf/libX11.so.6 (0xb60c7000)
	libxkbcommon-x11.so.0 => /lib/arm-linux-gnueabihf/libxkbcommon-x11.so.0 (0xb60b0000)
	libxkbcommon.so.0 => /lib/arm-linux-gnueabihf/libxkbcommon.so.0 (0xb6067000)
	libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb6054000)
	libstdc++.so.6 => /lib/arm-linux-gnueabihf/libstdc++.so.6 (0xb5f0d000)
	libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0xb5e8b000)
	libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0xb5e5e000)
	libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb5d10000)
	libxcb-xinput.so.0 => not found
	libexpat.so.1 => /lib/arm-linux-gnueabihf/libexpat.so.1 (0xb5ccf000)
	libuuid.so.1 => /tmp/.mount_Cura-mEqpI3J/usr/bin/qt/plugins/platforms/libuuid.so.1 (0xb5cb7000)
	/lib/ld-linux-armhf.so.3 (0xb6fa2000)
	libpng16.so.16 => /lib/arm-linux-gnueabihf/libpng16.so.16 (0xb5c7b000)
	libharfbuzz.so.0 => /lib/arm-linux-gnueabihf/libharfbuzz.so.0 (0xb5b7d000)
	libdbus-1.so.3 => /lib/arm-linux-gnueabihf/libdbus-1.so.3 (0xb5b2c000)
	libatomic.so.1 => /lib/arm-linux-gnueabihf/libatomic.so.1 (0xb5b13000)
	libicui18n.so.63 => /lib/arm-linux-gnueabihf/libicui18n.so.63 (0xb58c5000)
	libicuuc.so.63 => /lib/arm-linux-gnueabihf/libicuuc.so.63 (0xb573f000)
	libpcre2-16.so.0 => /lib/arm-linux-gnueabihf/libpcre2-16.so.0 (0xb56c3000)
	libdouble-conversion.so.1 => /lib/arm-linux-gnueabihf/libdouble-conversion.so.1 (0xb569f000)
	libglib-2.0.so.0 => /lib/arm-linux-gnueabihf/libglib-2.0.so.0 (0xb558f000)
	libGLdispatch.so.0 => /lib/arm-linux-gnueabihf/libGLdispatch.so.0 (0xb5510000)
	libXau.so.6 => /lib/arm-linux-gnueabihf/libXau.so.6 (0xb54fb000)
	libXdmcp.so.6 => /lib/arm-linux-gnueabihf/libXdmcp.so.6 (0xb54e6000)
	libgraphite2.so.3 => /lib/arm-linux-gnueabihf/libgraphite2.so.3 (0xb54b5000)
	libsystemd.so.0 => /lib/arm-linux-gnueabihf/libsystemd.so.0 (0xb5418000)
	libicudata.so.63 => /lib/arm-linux-gnueabihf/libicudata.so.63 (0xb3a1a000)
	libpcre.so.3 => /lib/arm-linux-gnueabihf/libpcre.so.3 (0xb39a3000)
	libbsd.so.0 => /lib/arm-linux-gnueabihf/libbsd.so.0 (0xb397b000)
	librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0xb3964000)
	liblzma.so.5 => /lib/arm-linux-gnueabihf/liblzma.so.5 (0xb3933000)
	liblz4.so.1 => /lib/arm-linux-gnueabihf/liblz4.so.1 (0xb3905000)
	libgcrypt.so.20 => /lib/arm-linux-gnueabihf/libgcrypt.so.20 (0xb382b000)
	libgpg-error.so.0 => /lib/arm-linux-gnueabihf/libgpg-error.so.0 (0xb3801000)
pi@raspber

**Comment by smartavionics** [comment]
Hi,Thanks for that. I can see that you are missing the xcb-xinput library on your system. I'm not near a computer right now to check but you need to install the libxcb-xinput package.Hope this helps! On 11 Feb 2022 08:33, daizhuo ***@***.***> wrote:
Thank you so much for ur help!
Finally I got these information.
***@***.***:~ $ ldd /tmp/.mount_Cura-mEqpI3J/usr/bin/qt/plugins/platforms/libqxcb.so
linux-vdso.so.1 (0xbeedf000)
/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so => /usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so (0xb6f8d000)
libQt5XcbQpa.so.5 => /tmp/.mount_Cura-mEqpI3J/usr/bin/qt/plugins/platforms/libQt5XcbQpa.so.5 (0xb6e7a000)
libfontconfig.so.1 => /tmp/.mount_Cura-mEqpI3J/usr/bin/qt/plugins/platforms/libfontconfig.so.1 (0xb6e30000)
libfreetype.so.6 => /lib/arm-linux-gnueabihf/libfreetype.so.6 (0xb6d77000)
libz.so.1 => /lib/arm-linux-gnueabihf/libz.so.1 (0xb6d4c000)
libQt5Gui.so.5 => /lib/arm-linux-gnueabihf/libQt5Gui.so.5 (0xb68cc000)
libQt5DBus.so.5 => /lib/arm-linux-gnueabihf/libQt5DBus.so.5 (0xb6833000)
libQt5Core.so.5 => /lib/arm-linux-gnueabihf/libQt5Core.so.5 (0xb63b9000)
libGLESv2.so.2 => /lib/arm-linux-gnueabihf/libGLESv2.so.2 (0xb639a000)
libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb6370000)
libX11-xcb.so.1 => /lib/arm-linux-gnueabihf/libX11-xcb.so.1 (0xb635e000)
libxcb-icccm.so.4 => /lib/arm-linux-gnueabihf/libxcb-icccm.so.4 (0xb634a000)
libxcb-image.so.0 => /lib/arm-linux-gnueabihf/libxcb-image.so.0 (0xb6336000)
libxcb-shm.so.0 => /lib/arm-linux-gnueabihf/libxcb-shm.so.0 (0xb6323000)
libxcb-util.so.0 => /lib/arm-linux-gnueabihf/libxcb-util.so.0 (0xb630e000)
libxcb-keysyms.so.1 => /lib/arm-linux-gnueabihf/libxcb-keysyms.so.1 (0xb62fb000)
libxcb-randr.so.0 => /lib/arm-linux-gnueabihf/libxcb-randr.so.0 (0xb62de000)
libxcb-render-util.so.0 => /lib/arm-linux-gnueabihf/libxcb-render-util.so.0 (0xb62ca000)
libxcb-render.so.0 => /lib/arm-linux-gnueabihf/libxcb-render.so.0 (0xb62af000)
libxcb-shape.so.0 => /lib/arm-linux-gnueabihf/libxcb-shape.so.0 (0xb629c000)
libxcb-sync.so.1 => /lib/arm-linux-gnueabihf/libxcb-sync.so.1 (0xb6286000)
libxcb-xfixes.so.0 => /lib/arm-linux-gnueabihf/libxcb-xfixes.so.0 (0xb626f000)
libxcb-xinerama.so.0 => /lib/arm-linux-gnueabihf/libxcb-xinerama.so.0 (0xb625c000)
libxcb-xkb.so.1 => /lib/arm-linux-gnueabihf/libxcb-xkb.so.1 (0xb6237000)
libxcb-xinput.so.0 => not found
libxcb.so.1 => /lib/arm-linux-gnueabihf/libxcb.so.1 (0xb6208000)
libXext.so.6 => /lib/arm-linux-gnueabihf/libXext.so.6 (0xb61e9000)
libX11.so.6 => /lib/arm-linux-gnueabihf/libX11.so.6 (0xb60c7000)
libxkbcommon-x11.so.0 => /lib/arm-linux-gnueabihf/libxkbcommon-x11.so.0 (0xb60b0000)
libxkbcommon.so.0 => /lib/arm-linux-gnueabihf/libxkbcommon.so.0 (0xb6067000)
libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb6054000)
libstdc++.so.6 => /lib/arm-linux-gnueabihf/libstdc++.so.6 (0xb5f0d000)
libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0xb5e8b000)
libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0xb5e5e000)
libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb5d10000)
libxcb-xinput.so.0 => not found
libexpat.so.1 => /lib/arm-linux-gnueabihf/libexpat.so.1 (0xb5ccf000)
libuuid.so.1 => /tmp/.mount_Cura-mEqpI3J/usr/bin/qt/plugins/platforms/libuuid.so.1 (0xb5cb7000)
/lib/ld-linux-armhf.so.3 (0xb6fa2000)
libpng16.so.16 => /lib/arm-linux-gnueabihf/libpng16.so.16 (0xb5c7b000)
libharfbuzz.so.0 => /lib/arm-linux-gnueabihf/libharfbuzz.so.0 (0xb5b7d000)
libdbus-1.so.3 => /lib/arm-linux-gnueabihf/libdbus-1.so.3 (0xb5b2c000)
libatomic.so.1 => /lib/arm-linux-gnueabihf/libatomic.so.1 (0xb5b13000)
libicui18n.so.63 => /lib/arm-linux-gnueabihf/libicui18n.so.63 (0xb58c5000)
libicuuc.so.63 => /lib/arm-linux-gnueabihf/libicuuc.so.63 (0xb573f000)
libpcre2-16.so.0 => /lib/arm-linux-gnueabihf/libpcre2-16.so.0 (0xb56c3000)
libdouble-conversion.so.1 => /lib/arm-linux-gnueabihf/libdouble-conversion.so.1 (0xb569f000)
libglib-2.0.so.0 => /lib/arm-linux-gnueabihf/libglib-2.0.so.0 (0xb558f000)
libGLdispatch.so.0 => /lib/arm-linux-gnueabihf/libGLdispatch.so.0 (0xb5510000)
libXau.so.6 => /lib/arm-linux-gnueabihf/libXau.so.6 (0xb54fb000)
libXdmcp.so.6 => /lib/arm-linux-gnueabihf/libXdmcp.so.6 (0xb54e6000)
libgraphite2.so.3 => /lib/arm-linux-gnueabihf/libgraphite2.so.3 (0xb54b5000)
libsystemd.so.0 => /lib/arm-linux-gnueabihf/libsystemd.so.0 (0xb5418000)
libicudata.so.63 => /lib/arm-linux-gnueabihf/libicudata.so.63 (0xb3a1a000)
libpcre.so.3 => /lib/arm-linux-gnueabihf/libpcre.so.3 (0xb39a3000)
libbsd.so.0 => /lib/arm-linux-gnueabihf/libbsd.so.0 (0xb397b000)
librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0xb3964000)
liblzma.so.5 => /lib/arm-linux-gnueabihf/liblzma.so.5 (0xb3933000)
liblz4.so.1 => /lib/arm-linux-gnueabihf/liblz4.so.1 (0xb3905000)
libgcrypt.so.20 => /lib/arm-linux-gnueabihf/libgcrypt.so.20 (0xb382b000)
libgpg-error.so.0 => /lib/arm-linux-gnueabihf/libgpg-error.so.0 (0xb3801000)
***@***.***

**Reply by daizhuo** [reply]
Wow…..I solved this problem with ur help! It works finally. Why is this apt missing? It is not a must require for the Pi OS? Anyway, Thank you so much 🙏
